src/main.ts

```diff
@@ -506,6 +517,52 @@ ipcMain.handle('app:open-log', () => {
   openLogFolder()
 })
 
+ipcMain.handle('settings:get', async () => {
+  console.info('settings:get event received')
+  return await getSettings()
+})
+
+ipcMain.handle('settings:save', async (event, data: AppSettings) => {
+  console.info('settings:save event received')
+
+  const browserWindow = browserWindowFromEvent(event)
+  try {
+    const diff = await saveSettings(data)
+    applySettings(diff, browserWindow)
```

Comment:
Event handler for saving settings from the UI. `saveSettings` returns the diff with only settings that have been modified. `applySettings` is responsible for overwriting and performing specific changes.